core/modes/sleep_mode.talon

```diff
@@ -36,3 +36,7 @@ settings():
     user.talon_mode()
 
 ^(talon wake)+$: speech.enable()
+
+# Catch the redundant sleep command, and notify the user that Talon is already asleep.
+# For a note about the optional <phrase>, see to_sleep_mode.talon.
+^(talon | go to) sleep [<phrase>]$: app.notify("Talon is already asleep")
```

Comment:
After taking a fresh look at this, I wonder if it's worth taking some time to either wrap these behavior in an action that we can override or override talon's actions based on the active mode. This would have the advantage that we could maybe define the commands in one place, potentially making it easier for folks to customize the commands and not miss that they exist in additional files.

Reply:
Yeah I agree. Would be better for spoken form to be defined in one place

Reply:
Sure yeah, I agree that sounds useful.
I don't really know how to do that though :sweat_smile:

Reply:
I took a quick crack at making this change. Please review/test when you can

Reply:
Huh ok, I think I understand. So you're overriding it closer to the source. It's easier to reason about.
I'm going to try test this now

Reply:
Yup, works fine for me. I don't use Dragon, so can't confirm those changes. But the regular sleep/wake works fine, and I get notified if Talon is already in the target mode.